### PR TITLE
[WIP] Add search functions to xccdf-guide-impl.xsl

### PR DESCRIPTION
**Pull request by redhatrises**
Adds similar functionality from xccdf-report-impl.xsl to search rules.

### CODE REVIEW

File: xsl/xccdf-guide-impl.xsl [hunk 1]
```diff
@@ -597,6 +597,18 @@ Authors:
     <div id="guide-tree">
```
Comment (matejak) [inline xsl/xccdf-guide-impl.xsl]: There are five opening `div` tags, but only four closing ones until the end of the file.

**Comment by mpreisler** [comment]
The javascript function shouldn't be copied like this, we should instead change the ruleSearch function to accomodate both cases. Copying code makes it less maintainable.

**Comment by yuumasato** [comment]
@openscap-ci test this please

**Comment by matejak** [comment]
Tests of `oscap` on RHEL6 are broken as of 3/2018, so the failure has nothing to do with the PR itself.

**Comment by matejak** [comment]
@openscap-ci test this please

**Comment by matejak** [comment]
Now, `make check` seems to be failing on an item that is related to the PR (it fails in `tests/API/XCCDF/unittests` and `tests/API/XCCDF/guide`).

**Comment by yuumasato** [comment]
@redhatrises Hello,

Can the search built in browser resolve the need for a search function?
Maybe more meaningful way to search rule is through IDs, and we can just expose them.

**Comment by yuumasato** [comment]
By clicking "Show all result detail" the rule IDs are shown and one can search using the browser search function.

**Comment by redhatrises** [comment]
Sorry, but this is about syncing our reports and guides formats to have similar workflows, reopening.

**Comment by jan-cerny** [comment]
The code here doesn't work, it needs to implement new functions in our JavaScript modules. @redhatrises We have discussed this with the team and we think that it's not necessary to keep this pull request. If you prefer, please open an issue.

**Pull request closed without merging**